### PR TITLE
build: use SQLite instead of CSV for skus, ~4x speed up

### PR DESCRIPTION
**Pull request by rominf**
First off, thanks for taking the time to contribute!

## Please Complete the Following

- [x] I read `CONTRIBUTING.md`
- [x] I used tabs to indent
- [x] I have tested my change

## Notes

Supersede #119.

Pipeline:
* `skus.sh` exports SKUs to SQLite database `skus.db` and does mapping.
* `pricing.pl` uses `skus.db` to build `pricing.yml`.

Separator in `mapping.csv` was changed to `,` to be able to use an official SQLite csv extension: https://www.sqlite.org/forum/info/b1d479097250d036

Also, fix description for `gce.network.internet.egress*` according to API changes.

### CODE REVIEW

**Comment by rominf** [comment]
Next step (PR):
1. Rewrite `build/skus.pl` to Go, leveraging https://pkg.go.dev/google.golang.org/api/cloudbilling/v1. It should improve type safety, robustness, and speed.
2. Combine it with `build/mapping.go` and name it `build/skus.go`.

**Comment by rominf** [comment]
I rebased on `master` and put all SQL into separate file to facilitate easier editing and make it more readable.

**Comment by rominf** [comment]
Nils, I noticed you rely on `sqlite` binary. If so, we can rely on it in `mapping.go` as well and switch to more popular https://github.com/mattn/go-sqlite3. What do you think?

**Comment by Cyclenerd** [comment]
The improvement in run time is impressive!

NEW:

```text
./mapping  0.64s user 0.06s system 86% cpu 0.802 total
perl pricing.pl  1.55s user 0.74s system 94% cpu 2.419 total
```

OLD:

```text
perl mapping.pl  110.62s user 2.31s system 99% cpu 1:53.15 total
perl pricing.pl  37.34s user 0.57s system 99% cpu 37.920 total
```

I have made a few changes (`skus.sql, `skus.sh`, `skus.pl`). For safety reasons, the database is now always newly created. To prevent Google from simply introducing new SKUs for the same service.

Please take a look @rominf . If you would like to make any changes, please feel free to do so and give me some feedback. The pull then get merged.

File: build/README.md [hunk 1]
```diff
@@ -95,64 +97,53 @@ bash skus.sh
...
-```bash
-{
-  cat "skus_compute.csv"
-  cat "skus_networking.csv"
-  cat "skus_storage.csv"
-  cat "skus_stackdriver.csv"
-  cat "skus_sql.csv"
-} > skus.csv
+perl skus.pl -id="9662-B51E-5089"
 ```
```
Comment (rominf) [inline build/README.md:121]: This should be dropped, since there will be no DB.

**Comment by rominf** [comment]
@Cyclenerd Thanks! I will remove Go for a while (until next PR) then. Will push changes in a few hours.

File: build/README.md (hunk 1 above)
Comment (Cyclenerd) [reply build/README.md]: Yes, you're right. A bit confusing. Text removed.

File: build/skus.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Create SQLite3 database for SKU export
+ */
+
+DROP TABLE IF EXISTS "skus";
```
Comment (rominf) [inline build/skus.sql:5]: I replaced `'` to `"`, see: https://www.sqlite.org/lang_keywords.html

File: build/README.md [hunk 3]
```diff
@@ -62,7 +61,7 @@ More help: <https://cloud.google.com/billing/v1/how-tos/catalog-api>
 	1. Select **Create credentials**, then select **API key** from the dropdown menu.
 	1. Copy your key and keep it secure.
 
-### 2️⃣  Export SKUs  adn add custom mapping (`skus.sh` and `skus.pl`)
+### 2️⃣  Export SKUs and add custom mapping (`skus.sh` and `skus.pl`)
```
Comment (rominf) [inline build/README.md:64]: :smile: :+1:

File: build/README.md [hunk 4]
```diff
@@ -28,28 +28,27 @@ Configuration files and scripts for generating the YAML file `pricing.yml` with
...
+ +----------------------------------+  +-----------------------------+
+                \                             /
+         +--------------------------------------------------+
+         | » Generate pricing information file (pricing.pl) |
+         +--------------------------------------------------+
+                              ↓
+                +-------------------------------+
+                |  GCP pricing information file |
+                |          (pricing.yml)        |
+                +-------------------------------+
```
Comment (rominf) [inline build/README.md:51]: Thanks!

File: build/mapping.sql [hunk 5]
```diff
@@ -41,3 +41,4 @@ SET "MAPPING" = (
 		"skus"."rowid" = "ranked_mapping"."skus_rowid"
 		AND "ranked_mapping"."rn" = 1
 );
+CREATE INDEX IF NOT EXISTS "pricing_index" ON "skus" ("MAPPING", "REGIONS") WHERE "MAPPING" IS NOT NULL;
```
Comment (rominf) [inline build/mapping.sql:44]: I think it's better to create `UNIQUE` index (it works faster and enforces correctness).

Comment (Cyclenerd) [reply build/mapping.sql:44]: It does not work because there are duplicate mappings for some regions (i.e. old 'Japan' region and new 'Tokyo' region).

**Comment by Cyclenerd** [comment]
Thank you very much for your work @rominf. I really like it!

GitHub Action run time:

OLD: 5m 27s
NEW: 2m 8s

I was able to successfully test it locally (without extra delay) and the GitHub action also executed successfully:
https://github.com/Cyclenerd/google-cloud-pricing-cost-calculator/commit/ebafa49d2b1962734f9a95eea3c4ea97c04dd381

Thank you very much again!

**Pull request merged**